inc/dashboard/filter.class.php

```diff
@@ -255,4 +258,47 @@ static function field(
 
       return $html.$js;
    }
+
+   /**
+    * Return filters for the provided dashboard
+    *
+    * @param int $dashboards_id
+    *
+    * @return array the JSON representation of the filter data
+    */
+   static function getForDashboard(int $dashboards_id = 0): string {
+      global $DB;
+
+      $dr_iterator = $DB->request([
+         'FROM'  => self::getTable(),
+         'WHERE' => [
+            'dashboards_dashboards_id' => $dashboards_id,
+         ]
+      ]);
+
+      $settings = $dr_iterator->next()['filter'];
```

Comment:
Unless I am wrong, filter can be missing in DB. For instance, migration does not create a filter for every existing dashboard.
```suggestion
      $settings = $dr_iterator->count() === 1 ? $dr_iterator->next()['filter'] : null;
```